### PR TITLE
Add "NetworkBalance" pie chart visualizer.

### PR DESCRIPTION
**Pull request by geoknee**
There is a deploy preview to check out (netlify / storybook) below.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,5 +46,8 @@
   "packageManager": "yarn@3.2.1",
   "engines": {
     "node": ">=16.0.0"
+  },
+  "dependencies": {
+    "@material-ui/core": "^4.12.4"
```
Comment (NiloCK) [inline package.json]: Am I reading correctly that this is in the base repo `package.json`? Is that possibly in error? (ie, intended for `./packages/site/package.json`).

File: packages/site/src/components/NetworkBalance.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+// This prevents a runtime error in storybook
+// @ts-ignore
+BigInt.prototype.toJSON = function (): string {
+  return this.toString();
+};
+
+export type NetworkBalanceProps = {
+  myBalanceFree: bigint;
+  myBalanceLocked: bigint;
+  hubBalanceFree: bigint;
```
Comment (NiloCK) [inline packages/site/src/components/NetworkBalance.tsx]: `theirBalanceFree`? `counterPartyBalanceFree`?

Comment (NiloCK) [reply packages/site/src/components/NetworkBalance.tsx]: (Not urgent)

File: packages/site/src/components/NetworkBalance.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        value: hubBalanceLockedPercentage,
+        color: '#d5dbe360',
+      },
+      {
+        title: prettyPrintWei(hubBalanceFree),
+        value: hubBalanceFreePercentage,
+        color: '#d5dbe3',
+      },
+    ];
+  }
```
Comment (NiloCK) [inline packages/site/src/components/NetworkBalance.tsx]:
```suggestion
const colors = {
  mine: '#ea692b',
  mineLocked: '#ea692b60',
  hub: '#d5dbe360',
  hubLocked: '#d5dbe3',
};

export const NetworkBalance: React.FC<NetworkBalanceProps> = (props) => {
  const { myBalanceFree, myBalanceLocked, hubBalanceFree, hubBalanceLocked } =
    props;
  const total =
    myBalanceFree + myBalanceLocked + hubBalanceFree + hubBalanceLocked;
  var data = [{ title: '0', value: 100, color: 'red' }];
  let myBalanceFreePercentage,
    myBalanceLockedPercentage,
    hubBalanceFreePercentage,
    hubBalanceLockedPercentage;
  if (total > 0) {
    [
      myBalanceFreePercentage,
      myBalanceLockedPercentage,
      hubBalanceFreePercentage,
      hubBalanceLockedPercentage,
    ] = [myBalanceFree, myBalanceLocked, hubBalanceFree, hubBalanceLocked].map(
      (x) => percentageOfTotal(x, total),
    );
    data = [
      {
        title: prettyPrintWei(myBalanceFree),
        value: myBalanceFreePercentage,
        color: colors.mine,
      },
      {
        title: prettyPrintWei(myBalanceLocked),
        value: myBalanceLockedPercentage,
        color: colors.mineLocked,
      },

      {
        title: prettyPrintWei(hubBalanceLocked),
        value: hubBalanceLockedPercentage,
        color: colors.hub,
      },
      {
        title: prettyPrintWei(hubBalanceFree),
        value: hubBalanceFreePercentage,
        color: colors.hubLocked,
      },
    ];
  }
```

Comment (NiloCK) [reply packages/site/src/components/NetworkBalance.tsx]: Still a duplication of the defined scss color in `_variables.scss`. Can we import from there?

**Review by NiloCK — APPROVED** [review]
Great :)

File: package.json (hunk 1 above)
Comment (geoknee) [reply package.json]: Yes I think you might be right. Will resolve.

Comment (geoknee) [reply package.json]: [694fc66](https://github.com/statechannels/schnap/pull/4/commits/694fc66ae958a9d55cc1ab6b1c386f5785d51201)

File: packages/site/src/components/NetworkBalance.tsx (hunk 2 above)
Comment (geoknee) [reply packages/site/src/components/NetworkBalance.tsx]: [c91c3dc](https://github.com/statechannels/schnap/pull/4/commits/c91c3dc7b003319b7919d60f91ffc2e16b70f3d0)

File: packages/site/src/components/NetworkBalance.tsx (hunk 3 above)
Comment (geoknee) [reply packages/site/src/components/NetworkBalance.tsx]: [dd35a55](https://github.com/statechannels/schnap/pull/4/commits/dd35a555bdbcb75e2508ed4e784105fad63b874b)

**Pull request merged**